lib/puppet/util/servicenow.rb

```diff
@@ -120,6 +123,36 @@ def calculate_event_conditions(resource_statuses)
   end
   module_function :calculate_event_conditions
 
+  def calculate_event_severity(resource_statuses, settings_hash)
+    # https://docs.servicenow.com/bundle/paris-it-operations-management/page/product/event-management-operator/concept/operator-events-alerts.html
+    # 0 => Clear....(The alert no longer needs action.)
+    # 1 => OK.......(No severity. An alert is created. The resource is still functional.)
+    # 2 => Warning..(Attention is required, even though the resource is still functional.)
+    # 3 => Minor....(Partial, non-critical loss of functionality or performance degradation occurred.)
+    # 4 => Major....(Major functionality is severely impaired or performance has degraded.)
+    # 5 => Critical.(The resource is either not functional or critical problems are imminent.)
+    # Loop through each event condition and append it's corresponding severity
+    event_conditions = calculate_event_conditions(resource_statuses)
+    severities = []
+    event_conditions.each do |condition, exists|
+      next unless exists
+      case condition
+      when 'failures'
+        severities << settings_hash['failures_event_severity']
+      when 'corrective_changes'
+        severities << settings_hash['corrective_change_event_severity']
```

Comment:
If the name of the setting and thus the name of the hash key here was changed for each of these so that the `s` agreed with the case statement case, then this case statement could probably go away. For example if `corrective_change_event_severity` became `corrective_changes_event_severity`, then this could become something more like:
```
event_conditions.select{|_, exists| exists == true}.map {|condition, _| settings_hash[condition]}.sort.first
```
or something like that.

Reply:
ah yeah that would make things simpler.

Reply:
updated.